### DISCUSSION TITLE
Add blade @asset and @route

### DISCUSSION THREAD

**Discussion by akbarali1** [laravel/framework#50398] [Ideas]
Now in blade it should be used in `{{ asset('coreui/css/style.css') }}` method. 
Can you redo it and add `@asset('coreui/css/style.css')` ?

**Comment by akbarali1** [comment]
I don't see any problem with that. Or are there any problems?

**Reply by akbarali1** [reply]
I also think that `@route('home')` should be added.

**Reply by Raza9798** [reply]
better to explain more about why you requesting this changes

**Reply by akbarali1** [reply]
I use this `{{ asset('css/style.css') }}` a lot in Blade. route() is used more like this. It would be a bit more convenient if `@route()` was `@asset()`

**Reply by Raza9798** [reply]
you can solve this using `AppServiceProvider` on your project

```
<?php

namespace App\Providers;

use Illuminate\Support\Facades\Blade;
use Illuminate\Support\ServiceProvider;

class AppServiceProvider extends ServiceProvider
{
    /**
     * Bootstrap any application services.
     *
     * @return void
     */
    public function boot()
    {
        // Define a custom Blade directive named '@asset'
        Blade::directive('asset', function ($path) {
            return "<?php echo asset($path); ?>";
        });
    }

    /**
     * Register any application services.
     *
     * @return void
     */
    public function register()
    {
        //
    }
}
```

Now, you can use the `@asset` directive in your Blade templates to generate asset URLs. as follows
```
<link rel="stylesheet" href="@asset('coreui/css/style.css')">
```

**Reply by akbarali1** [reply]
Thanks. 

I knew this method. But I think it's better if it comes with Laravel.
